### ISSUE TITLE
qthelp builder should htmlescape keywords

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1435]
Building qch documentation fails without this patch, as the keywords contain `'<CONFIG>'` etc:

 http://www.cmake.org/cmake/help/v3.0/variable/CMAKE_EXE_LINKER_FLAGS_CONFIG.html

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1435
- Originally reported by: Anonymous
- Originally created at: 2014-03-31T15:02:05.418

**Comment by tk0miya** [comment]
Here is an attached patch (salvaged from bitbucket):
```
--- sphinx/builders/qthelp.py   2014-03-31 14:56:20.683148694 +0200
+++ /usr/lib/python2.7/dist-packages/sphinx/builders/qthelp.py  2014-03-31 14:56:01.055149515 +0200
@@ -266,9 +266,9 @@

         if id:
             item = ' '*12 + '<keyword name="%s" id="%s" ref="%s"/>' % (
-                name, id, ref[1])
+                name, id, htmlescape(ref[1]))
         else:
-            item = ' '*12 + '<keyword name="%s" ref="%s"/>' % (name, ref[1])
+            item = ' '*12 + '<keyword name="%s" ref="%s"/>' % (name, htmlescape(ref[1]))
         item.encode('ascii', 'xmlcharrefreplace')
         return item

@@ -287,7 +287,7 @@
 #                                                        title, i, ref))
 #                item.encode('ascii', 'xmlcharrefreplace')
 #                keywords.append(item)
-                keywords.append(self.keyword_item(title, ref))
+                keywords.append(self.keyword_item(title, htmlescape(ref)))

         if subitems:
             for subitem in subitems:
```

**Comment by tk0miya** [comment]
Fixed by #4749.
Thank you for reporting
